.github/actions/sil-kit-ci/setup/action.yml

```diff
@@ -0,0 +1,77 @@
+name: sil-kit-ci/setup
+
+description: Setup the environment for the other CI actions.
+
+inputs:
+
+  preset-name:
+    description: The preset name from CMakePresets.json.
+    required: true
+
+  linux-docker-image:
+    description: Docker image used under Linux
+    required: true
+
+  msvc-arch:
+    description: Target architecture used under Windows/MSVC
+    required: true
+
+  msvc-toolset:
+    description: Target toolset used under Windows/MSVC
+    required: true
+
+runs:
+  using: composite
+  steps:
+
+    - name: compute directories
+      uses: ./.github/actions/sil-kit-ci/dirs
+      id: dirs
+      with:
+        preset-name: ${{ inputs.preset-name }}
+
+    - name: (all) checkout sil-kit sources
+      uses: actions/checkout@v4
+      with:
+        path: ${{ steps.dirs.outputs.cmake-source }}
+        submodules: true
+
+    - name: (linux) cleanup build and package directories
```

Comment:
It should be easily possible to simplify all of this Linux/Windows duplicates using `bash` on Linux and Windows runners.

Reply:
On second though, we might run into path separator issues though.

Reply:
Which should now be fine, after switching the paths over to forward slashes 👍